azext_edge/tests/edge/init/test_permissions_unit.py

```diff
@@ -0,0 +1,113 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+import pytest
+
+from azext_edge.edge.providers.orchestration.permissions import verify_write_permission_against_rg
+from azure.cli.core.azclierror import ValidationError
+
+from ...generators import get_zeroed_subscription, generate_generic_id
+
+MOCK_SUBSCRIPTION_ID = get_zeroed_subscription()
+MOCK_RG = f"rg_{generate_generic_id()}"
+
+
+class IntermObj:
```

Comment:
assuming this is to stop having .return_value.as_dict.return_value -> we could have this as a general object for tests

Reply:
yea , works for me